### ISSUE TITLE
truncated decryption

### ISSUE THREAD

**Issue by grantmacken** [eXist-db/crypto-exist-java-lib#7]
I am getting a truncated decrypted string when using either AES or DES

```
<query xmlns="http://exist.sourceforge.net/NS/exist"
 start='1'
 max='9999'
 wrap="no">
<text>
<![CDATA[
xquery version "3.1";
import module namespace crypto="http://expath.org/ns/crypto";
try{
let $msg1 := ' testing 12345'
let $msg2 := ' testing 12345 abcde truncated'
let $type := 'symmetric'
let $key :=  'keykeykekeykeyke'
let $algorithm := 'AES'
let $iv := crypto:hash('initialization vector', 'MD5', 'base64')
let $provider := 'SunJCE'
let $encrypted1 := crypto:encrypt($msg1,$type,$key,$algorithm,(),() )
let $decypted1 := crypto:decrypt($encrypted1, $type, $key,$algorithm,(),())
let $encrypted2 := crypto:encrypt($msg2,$type,$key,$algorithm,(),() )
let $decypted2 := crypto:decrypt($encrypted2, $type, $key,$algorithm,(),())
return
(
'encrypted 1: ' , $encrypted1 ,'&#10;',
'decypted 1: ' , $decypted1 ,'&#10;',
'encrypted 2: ' , $encrypted2 ,'&#10;',
'decypted 2: ' , $decypted2 ,'&#10;'
)
} catch * {(
 $err:code || ': '  || $err:description || '&#10;'
)}
]]>
</text>
</query>
```
The query results are

```
encrypted 1: 214-175-28-56-201-44-14-203-187-91-223-121-53-252-213-176
decypted 1:  testing 12345 
encrypted 2: 54-108-156-133-204-164-215-123-137-112-5-88-37-88-36-4
decypted 2: bcde truncated 
```

The longer string is being encrypted but is not being decrypted properly

**Comment by claudius108** [comment]
I reproduced the error, I will try to fix it.

```xquery
xquery version "3.1";

import module namespace crypto="http://expath.org/ns/crypto";

try{
let $msg1 := '12345678901234567890'
let $msg2 := ' testing 12345'
let $type := 'symmetric'
let $key :=  'keykeykekeykeyke'
let $algorithm := 'AES'

let $encrypted1 := crypto:encrypt($msg1, $type, $key, $algorithm, (), ())
let $decypted1 := crypto:decrypt($encrypted1, $type, $key, $algorithm, (), ())
let $encrypted2 := crypto:encrypt($msg2, $type, $key, $algorithm, (), ())
let $decypted2 := crypto:decrypt($encrypted2, $type, $key,$algorithm, (), ())
return
(
'encrypted 1: ' || $encrypted1,
'decypted 1: ' || $decypted1,
'encrypted 2: ' || $encrypted2,
'decypted 2: ' || $decypted2
)
} catch * {(
 $err:code || ': '  || $err:description || '&#10;'
)}
xquery

**Comment by claudius108** [comment]
I reproduced the error, I will try to fix it.

```xquery
xquery version "3.1";

import module namespace crypto="http://expath.org/ns/crypto";

try{
let $msg1 := '12345678901234567890'
let $msg2 := ' testing 12345'
let $type := 'symmetric'
let $key :=  'keykeykekeykeyke'
let $algorithm := 'AES'

let $encrypted1 := crypto:encrypt($msg1, $type, $key, $algorithm, (), ())
let $decypted1 := crypto:decrypt($encrypted1, $type, $key, $algorithm, (), ())
let $encrypted2 := crypto:encrypt($msg2, $type, $key, $algorithm, (), ())
let $decypted2 := crypto:decrypt($encrypted2, $type, $key,$algorithm, (), ())
return
(
'encrypted 1: ' || $encrypted1,
'decypted 1: ' || $decypted1,
'encrypted 2: ' || $encrypted2,
'decypted 2: ' || $decypted2
)
} catch * {(
 $err:code || ': '  || $err:description || '&#10;'
)}
xquery

**Comment by claudius108** [comment]
Hi,

Version 0.7 of the module solves this, see the eXist public repo.

**Comment by line-o** [comment]
Confirmed fixed (tested using eXist-5.3.0 and EXPath-crypto-lib 6.0.0)
